src/ert/storage/local_storage.py

```diff
@@ -212,8 +244,73 @@ def create_ensemble(
         self._ensembles[ens.id] = ens
         return ens
 
+    def _save_index(self) -> None:
+        if not hasattr(self, "_index"):
+            return
+
+        with open(self.path / "index.json", mode="w", encoding="utf-8") as f:
+            print(self._index.json(), file=f)
+
     def _load_experiment(self, uuid: UUID) -> LocalExperimentAccessor:
         return LocalExperimentAccessor(self, uuid, self._experiment_path(uuid))
 
     def _load_ensemble(self, path: Path) -> LocalEnsembleAccessor:
         return LocalEnsembleAccessor(self, path, refcase=self._refcase)
+
+
+def local_storage_needs_migration(path: os.PathLike[str]) -> bool:
+    """
+    Checks whether the path points to a LocalStorage that is in need of
+    migration. Returns true if the LocalStorage is outdated OR if the path does
+    not point to a LocalStorage directory (eg. it is a URL that points to a ERT
+    Storage Server)
+    """
+    path = Path(path)
+
+    if not path.exists():
+        return False
+
+    try:
+        with open("index.json", encoding="utf-8") as f:
```

Comment:
Is it perhaps better to use `_Index.parse_file` here as well?

Reply:
It might not be. We are only interested in the `version` key because the structure of `index.json` is unknown at this time. `_Index.parse_file` would imply that the structure of `index.json` is the same as it is currently, which may not be the case in the future.